cmd/xl-v1-healing-common.go

```diff
@@ -183,10 +183,10 @@ func disksWithAllParts(ctx context.Context, onlineDisks []StorageAPI, partsMetad
 			// it needs healing too.
 			for _, part := range partsMetadata[i].Parts {
 				checksumInfo := erasureInfo.GetChecksumInfo(part.Name)
-				err = onlineDisk.VerifyFile(bucket, pathJoin(object, part.Name), checksumInfo.Algorithm, checksumInfo.Hash, erasure.ShardSize())
+				err = onlineDisk.VerifyFile(bucket, pathJoin(object, part.Name), part.Size == 0, checksumInfo.Algorithm, checksumInfo.Hash, erasure.ShardSize())
```

Comment:
@vadmeste 

`erasure.ShardFileSize(part.Size)` will give you exact shard file size, hence you can use that instead of `empty` boolean

Reply:
> @vadmeste
> 
> `erasure.ShardFileSize(part.Size)` will give you exact shard file size, hence you can use that instead of `empty` boolean

Well, bitrot works well for any size, but it just doesn't know what to do for empty files.. I am not sure we need to use size instead of empty flag, (this is the PR which does that anyway https://github.com/minio/minio/pull/7932)

Reply:
`empty` files are not possible practically since we use O_DIRECT. Similarly incorrect sizes are also not possible. (unless the backend FS does not support O_DIRECT)

Instead of using empty bool you can use the size itself as it is available. For the code reader it is more obvious to see that we are verifying exact size instead of checking for `empty`